mantis-server/mantis-server-agent/src/main/java/io/mantisrx/server/agent/metrics/cgroups/CgroupImpl.java

```diff
@@ -22,43 +22,109 @@
 import java.nio.file.Files;
 import java.nio.file.Path;
 import java.nio.file.Paths;
+import java.util.Arrays;
+import java.util.Collections;
+import java.util.List;
 import java.util.Map;
+import java.util.Objects;
 import java.util.stream.Collectors;
+import javax.annotation.Nullable;
+import lombok.AccessLevel;
+import lombok.Getter;
 import lombok.RequiredArgsConstructor;
 
 @RequiredArgsConstructor
 public class CgroupImpl implements Cgroup {
+
     private final String path;
 
+    private List<String> getSubsystems() {
+        return
+            Arrays.asList(Objects.requireNonNull(Paths.get(path).toFile().listFiles()))
+                .stream()
+                .filter(s -> s.isDirectory())
+                .map(s -> s.getName())
+                .collect(Collectors.toList());
+    }
+
+    /**
+     * Maybe change this to the below command in the future.
+     * <p>
+     * ``` stat -fc %T /sys/fs/cgroup/ ``` This should return cgroup2fs for cgroupv2 and tmpfs for
+     * cgroupv1.
+     */
+    @Getter(lazy = true, value = AccessLevel.PRIVATE)
+    private final boolean old = getSubsystems().size() > 0;
+
+    @Override
+    public Boolean isV1() {
+        return isOld();
+    }
+
+    @Override
+    public Boolean isV2() {
```

Comment:
This doesn't seem to be used at all, I think you can get rid of it and just get away with `isV1`

Reply:
good point. Removed.